impress-org/givewp
# Feature: add phone number to donor profile

Resolves [GIVE-531]

## Description



This PR adds a phone input to the donor profile page. The [International Telephone Input JS library (sponsored by Twilio)](https://github.com/jackocnr/intl-tel-input) was used to make sure that the numbers were stored in the proper format regardless of the donor country.

The big advantage of using this JS library is that we can validate the numbers for all countries before submitting the form and store the donor phone number using the international format - which can be useful in the future if we decide to implement some kind of integration with 3rd party software to send SMS to donors because services like that (For example Twilio) require numbers in the international format.

The whole functionality was encapsulated in the `give_get_intl_tel_input()` function, this way we can reuse it in other places that require a phone input using only vanilla JS. 

~~In a subsequent PR, we also can create a version of this phone input in our form builder library to allow the use as a React Component.~~  **Edit:** The library already provides a [React Component](https://intl-tel-input.com/examples/react-component.html) that we can use when working in the Visual Form Builder or any other UI that uses React.

## Affects



The donor profile page.

## Visuals



![image](https://github.com/impress-org/givewp/assets/16308864/f523bc64-7dfd-4fe0-8e14-ac3fa9257049)

![image](https://github.com/impress-org/givewp/assets/16308864/7d5717ac-5f36-4404-92cf-4e8b1a253593)


## Testing Instructions



1. Go to the donor profile page;
2. Edit the donor details;
3. Try to save an empty phone number, it should work;
4. Try to save a valid phone number, it should work;
5. Try to save an invalid phone number, it should NOT work.

## Pre-review Checklist






[GIVE-531]: https://stellarwp.atlassian.net/browse/GIVE-531?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
